barretenberg/cpp/src/barretenberg/eccvm_recursion/eccvm_recursive_verifier.cpp

```diff
@@ -30,6 +30,12 @@ template <typename Flavor> void ECCVMRecursiveVerifier_<Flavor>::verify_proof(co
     const auto circuit_size = transcript->template receive_from_prover<BF>("circuit_size");
     for (auto [comm, label] : zip_view(commitments.get_wires(), commitment_labels.get_wires())) {
         comm = transcript->template receive_from_prover<Commitment>(label);
+        // TODO(https://github.com/AztecProtocol/barretenberg/issues/1017): This is a hack to ensure zero commitments
+        // are still on curve as the transcript doesn't currently support a point at infinity representation for
+        // cycle_group
+        if (!comm.get_value().on_curve()) {
```

Comment:
If the point at infinity is at another place, this will fail the circuit. You need to do set_point_at_infinity(x==x_error&&y=y_error)